**PR #2152 -- eclipse-che/che-operator** -- fix: grant workspace users ImageStream read access

| Check | Category | Status | Duration | CI |
|-------|----------|--------|----------|----|
| ci/prow/v19-devworkspace-happy-path | E2E Tests | FAILED | — | Prow |
| ci/prow/v19-upgrade-stable-to-next | E2E Tests | FAILED | — | Prow |
| resources-validation | Validation | FAILED | 1m39s | GHA |
| Analyze (actions) | Static Analysis | PASSED | 57s | GHA |
| Analyze (go) | Static Analysis | PASSED | 4m52s | GHA |
| CodeQL | Static Analysis | PASSED | 4s | External |
| WIP | External | PASSED | — | External |
| add-web-ide-link | External | PASSED | 4s | External |
| bundle-version-validation | Validation | PASSED | 26s | GHA |
| coverage-report | Unit Tests | PASSED | 2m27s | GHA |
| dependencies-validation | Validation | PASSED | 26s | GHA |
| eclipsefdn/eca | External | PASSED | — | External |
| helm-on-minikube | Integration Tests | PASSED | 23m19s | GHA |
| multiplatform-image-build (ubuntu-24.04) | Build | PASSED | 3m46s | GHA |
| multiplatform-image-build (ubuntu-24.04-arm) | Build | PASSED | 2m53s | GHA |
| operator-on-minikube (ubuntu-24.04) | Integration Tests | PASSED | 19m14s | GHA |
| operator-on-minikube (ubuntu-24.04-arm) | Integration Tests | PASSED | 18m39s | GHA |
| source-code-validation | Static Analysis | PASSED | 3m7s | GHA |
| unit-tests | Unit Tests | PASSED | 2m18s | GHA |
| upgrade-stable-to-next-on-minikube | Integration Tests | PASSED | 22m59s | GHA |
| workspaces.openshift.com | External | PASSED | — | External |

---

### FAILED resources-validation (GHA)

The `resources-validation` check failed because generated resources are not up to date with the current code. The check detected differences in the generated YAML files and reported: `[ERROR] Resources are not up to date. [ERROR] Run 'make update-dev-resources' to update them.` This indicates that changes to the code require regenerating the resource files to match.

Log: [resources-validation](https://github.com/eclipse-che/che-operator/actions/runs/29009489493/job/86089244053)
Rerun: [Re-run failed jobs](https://github.com/eclipse-che/che-operator/actions/runs/29009489493)

---

### FAILED ci/prow/v19-devworkspace-happy-path (Prow)

The E2E test failed during Eclipse Che deployment when waiting for the operator to report a valid version. After deploying all required components (operator, dashboard, gateway, devworkspace-controller), the test tried to verify the `cheVersion` status field was set to "next" but found it empty. The error `[ERROR] Failed to deploy Eclipse Che '' version` indicates the CheCluster status never populated the expected version field, causing the test to timeout after 500 iterations. This appears to be a deployment/reconciliation issue rather than an infrastructure failure.

Log: [ci/prow/v19-devworkspace-happy-path](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/eclipse-che_che-operator/2152/pull-ci-eclipse-che-che-operator-main-v19-devworkspace-happy-path/2078055726411943936)
Rerun: Post `/test v19-devworkspace-happy-path` as a PR comment
  Or rerun all failed Prow jobs: `/retest`

---

### FAILED ci/prow/v19-upgrade-stable-to-next (Prow)

This upgrade test failed with the same root cause as the devworkspace-happy-path test. After successfully upgrading from stable to the "next" version, the test attempted to verify the upgraded deployment by checking the `cheVersion` status field. However, it found an empty version string instead of "next", resulting in `[ERROR] Failed to deploy Eclipse Che '' version`. The Eclipse Che operator pods were running, but the CheCluster status reconciliation failed to report the expected version, causing the test to fail after timeout.

Log: [ci/prow/v19-upgrade-stable-to-next](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/eclipse-che_che-operator/2152/pull-ci-eclipse-che-che-operator-main-v19-upgrade-stable-to-next/2078055726445498368)
Rerun: Post `/test v19-upgrade-stable-to-next` as a PR comment
  Or rerun all failed Prow jobs: `/retest`

--- a/bundle/next/eclipse-che/manifests/che-operator.clusterserviceversion.yaml
+++ b/bundle/next/eclipse-che/manifests/che-operator.clusterserviceversion.yaml
@@ -86,7 +86,7 @@ metadata:
     categories: Developer Tools
     certified: "false"
     containerImage: quay.io/eclipse/che-operator:next
-    createdAt: "2026-07-20T15:40:22Z"
+    createdAt: "2026-07-21T11:26:48Z"
     description: A Kube-native development solution that delivers portable and collaborative
       developer workspaces.
     features.operators.openshift.io/cnf: "false"
@@ -108,7 +108,7 @@ metadata:
     operatorframework.io/arch.amd64: supported
     operatorframework.io/arch.arm64: supported
     operatorframework.io/os.linux: supported
-  name: eclipse-che.v7.121.0-1041.next
+  name: eclipse-che.v7.121.0-1042.next
   namespace: placeholder
 spec:
   apiservicedefinitions: {}
@@ -690,6 +690,13 @@ spec:
               verbs:
                 - create
             - apiGroups:
+                - image.openshift.io
+              resources:
+                - imagestreams
+              verbs:
+                - get
+                - list
+            - apiGroups:
                 - ""
               resources:
                 - events
@@ -1166,7 +1173,7 @@ spec:
       name: openvsx
     - image: quay.io/sclorg/postgresql-16-c9s:20260319
       name: openvsx-postgres
-  version: 7.121.0-1041.next
+  version: 7.121.0-1042.next
   webhookdefinitions:
     - admissionReviewVersions:
         - v1

--- a/config/rbac/cluster_role.yaml
+++ b/config/rbac/cluster_role.yaml
@@ -162,6 +162,13 @@ rules:
     verbs:
       - create
   - apiGroups:
+      - image.openshift.io
+    resources:
+      - imagestreams
+    verbs:
+      - get
+      - list
+  - apiGroups:
       - ''
     resources:
       - events

--- a/deploy/deployment/kubernetes/combined.yaml
+++ b/deploy/deployment/kubernetes/combined.yaml
@@ -28124,6 +28124,13 @@ rules:
   verbs:
   - create
 - apiGroups:
+  - image.openshift.io
+  resources:
+  - imagestreams
+  verbs:
+  - get
+  - list
+- apiGroups:
   - ""
   resources:
   - events

--- a/deploy/deployment/kubernetes/objects/che-operator.ClusterRole.yaml
+++ b/deploy/deployment/kubernetes/objects/che-operator.ClusterRole.yaml
@@ -162,6 +162,13 @@ rules:
   verbs:
   - create
 - apiGroups:
+  - image.openshift.io
+  resources:
+  - imagestreams
+  verbs:
+  - get
+  - list
+- apiGroups:
   - ""
   resources:
   - events

--- a/deploy/deployment/openshift/combined.yaml
+++ b/deploy/deployment/openshift/combined.yaml
@@ -28124,6 +28124,13 @@ rules:
   verbs:
   - create
 - apiGroups:
+  - image.openshift.io
+  resources:
+  - imagestreams
+  verbs:
+  - get
+  - list
+- apiGroups:
   - ""
   resources:
   - events

--- a/deploy/deployment/openshift/objects/che-operator.ClusterRole.yaml
+++ b/deploy/deployment/openshift/objects/che-operator.ClusterRole.yaml
@@ -162,6 +162,13 @@ rules:
   verbs:
   - create
 - apiGroups:
+  - image.openshift.io
+  resources:
+  - imagestreams
+  verbs:
+  - get
+  - list
+- apiGroups:
   - ""
   resources:
   - events

--- a/helmcharts/next/templates/che-operator.ClusterRole.yaml
+++ b/helmcharts/next/templates/che-operator.ClusterRole.yaml
@@ -162,6 +162,13 @@ rules:
   verbs:
   - create
 - apiGroups:
+  - image.openshift.io
+  resources:
+  - imagestreams
+  verbs:
+  - get
+  - list
+- apiGroups:
   - ""
   resources:
   - events

--- a/helmcharts/next/templates/org_v2_checluster.yaml
+++ b/helmcharts/next/templates/org_v2_checluster.yaml
@@ -9,7 +9,6 @@
 # Contributors:
 #   Red Hat, Inc. - initial API and implementation
 #
-
 apiVersion: org.eclipse.che/v2
 kind: CheCluster
 metadata:

--- a/pkg/deploy/server/rbac.go
+++ b/pkg/deploy/server/rbac.go
@@ -280,6 +280,11 @@ func (s *CheServerReconciler) getUserCommonPolicies() []rbacv1.PolicyRule {
 			Resources: []string{"projects"},
 			Verbs:     []string{"get"},
 		},
+		{
+			APIGroups: []string{"image.openshift.io"},
+			Resources: []string{"imagestreams"},
+			Verbs:     []string{"get", "list"},
+		},
 	}
 
 	if infrastructure.IsOpenShift() {

